### PR TITLE
Classify non-code files as collaborative or non-collaborative with user verification. (Pt-5)

### PR DESCRIPTION
**Pull request by 6s-1**
## 📝 Description

This PR is part 5 which is the final part in a series of PRs for the Non-code File Checker file and adds comprehensive file classification with user verification functionality to the capstone project. Building on previous work, this PR introduces the main orchestrator function [classify_non_code_files_with_user_verification()] that brings together git detection, metadata collection, and user contribution verification into a single, reusable pipeline.

Key Features:

- Complete Classification Pipeline: New function that orchestrates git detection, user identity extraction, metadata collection, and file categorization
- User Verification Logic: Validates user contributions by checking git commit history against provided or system-detected user email
- Three-Category Classification:
-- Collaborative: Git repo files with 2+ authors where user is one of the contributors
-- Non-Collaborative: Git repo files with only the user as author OR all local files in non-git directories
-- Excluded: Git repo files where user is NOT a contributor (others' work)
-Seamless Git and Local Support: Handles both git repositories and local directories with appropriate classification logic for each
- Error Handling: Gracefully handles cases where git user identity cannot be determined
- Code Reuse: Leverages existing utilities from [git_utils.py], [scan_utils.py] (scan_project_files), and functions from this module.

This PR completes the core classification functionality needed for the non-code analysis pipeline and sets the foundation for document parsing and analysis in subsequent PRs.

**Closes:** #261 #260 

---

## 🔧 Type of Change

- [ ] 🐛 Bug fix (non-breaking change that fixes an issue)
- [x] ✨ New feature (non-breaking change that adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] 📚 Documentation added/updated
- [x] ✅ Test added/updated
- [x] ♻️ Refactoring
- [x] ⚡ Performance improvement

---

## 🧪 Testing

Comprehensive test coverage has been added for the new [classify_non_code_files_with_user_verification()] function with 23 new test cases covering all scenarios and edge cases. The tests validate the complete classification pipeline including git repository detection, user identity extraction, metadata collection, and three-category file classification (collaborative, non-collaborative, and excluded). Tests use mocking to isolate functionality and ensure fast, deterministic execution without requiring actual git repositories or file systems.

---

## ✓ Checklist

- [x] 🤖 GenAI was used in generating the code and I have performed a self-review of my own code
- [x] 💬 I have commented my code where needed
- [x] 📖 I have made corresponding changes to the documentation
- [x] ⚠️ My changes generate no new warnings
- [x] ✅ I have added tests that prove my fix is effective or that my feature works and tests are passing locally
- [x] 🔗 Any dependent changes have been merged and published in downstream modules

---

### CODE REVIEW

File: app/utils/non_code_analysis/non_code_file_checker.py [hunk 1]
```diff
@@ -238,4 +239,91 @@ def verify_user_in_files(
...
+        >>> print(f"Collaborative (you + others): {len(result['collaborative'])}")
+        >>> print(f"Non-collaborative (solo or local): {len(result['non_collaborative'])}")
+        
+        # Non-git directory (all files are non-collaborative)
+        >>> result = classify_non_code_files_with_user_verification('/path/to/local')
+        >>> print(f"All local files (non-collaborative): {len(result['non_collaborative'])}")
+    """
+    directory = Path(directory)
+    
+    # REUSED: detect_git() from git_utils.py
```
Comment (abstractafua) [inline app/utils/non_code_analysis/non_code_file_checker.py]: Great job leveraging re-usability here

File: app/utils/non_code_analysis/non_code_file_checker.py [hunk 2]
```diff
@@ -238,4 +239,91 @@ def verify_user_in_files(
...
+        if user_email is None:
+            user_identity = get_git_user_identity(directory)
+            user_email = user_identity.get("email", "")
+        else:
+            user_identity = {"name": "", "email": user_email}
+        
+        if not user_email:
+            # Can't determine user - treat as error case
+            return {
+                "is_git_repo": True,
```
Comment (abstractafua) [inline app/utils/non_code_analysis/non_code_file_checker.py]: @6s-1 Do you think it's worth populating this object if the user's email can't be confirmed? I'm thinking that the attribute `is_git_repo` = True 
may not provide us any value later down the line since this would still be considered an error case.

**Review by abstractafua — APPROVED** [review]
Great work Shreya! Left you a couple comments. I don't see any issues with your integration otherwise. Approved.

File: app/utils/non_code_analysis/non_code_file_checker.py (hunk 1 above)
Comment (6s-1) [reply app/utils/non_code_analysis/non_code_file_checker.py]: Thank you!

File: tests/utils/non_code_analysis/test_non_code_file_checker.py
[the remark this answers is not in the document]
Comment (6s-1) [reply tests/utils/non_code_analysis/test_non_code_file_checker.py]: Good catch on those edge cases! While duplicate user emails in contributor lists are unlikely (git typically deduplicates by design), they're worth documenting as known edge cases rather than adding explicit tests right now. Our current implementation using Python set() operations and in checks would handle duplicates naturally - if "alice@example.com" appears twice in a file's author list, verify_user_in_files() would still return True correctly.
